src/types/redis_zset.cc

```diff
@@ -52,24 +52,12 @@ rocksdb::Status ZSet::Add(const Slice &user_key, ZAddFlags flags, MemberScores *
   WriteBatchLogData log_data(kRedisZSet);
   batch->PutLogData(log_data.Encode());
   std::string member_key;
-  std::set<std::string> added_member_keys;
-  for (int i = static_cast<int>(mscores->size() - 1); i >= 0; i--) {
-    InternalKey(ns_key, (*mscores)[i].member, metadata.version, storage_->IsSlotIdEncoded()).Encode(&member_key);
-
-    // Fix the corner case that adds the same member which may add the score
-    // column family many times and cause problems in the ZRANGE command.
-    //
-    // For example, we add members with `ZADD mykey 1 a 2 a` and `ZRANGE mykey 0 1`
-    // return only one member(`a`) was expected but got the member `a` twice now.
-    //
-    // The root cause of this issue was the score key was composed by member and score,
-    // so the last one can't overwrite the previous when the score was different.
-    // A simple workaround was add those members with reversed order and skip the member if has added.
-    if (added_member_keys.find(member_key) != added_member_keys.end()) {
+  std::unordered_set<std::string_view> added_member_keys;
```

Comment:
So previously, `added_member_keys` is a "decoded" key, now, it becomes a "user-key"?

Reply:
Previously, the repeated members are filtered by encoded member_key, but using the original member_key can achieve the same effect.